activerecord/lib/active_record/fixtures.rb

```diff
@@ -986,7 +986,7 @@ def setup_fixtures(config = ActiveRecord::Base)
       # Load fixtures for every test.
       else
         ActiveRecord::FixtureSet.reset_cache
-        @@already_loaded_fixtures[self.class] = nil
+        @@already_loaded_fixtures = {}
         @loaded_fixtures = load_fixtures(config)
```

Comment:
I don't think we want to do that, since the fixtures loaded in the else statement are used in a non-transactional test and are not going to be rolled back before the next test runs.